### PR TITLE
Fan club role and role-based auth middleware

### PR DESCRIPTION
**Pull request by rado196**

### CODE REVIEW

**Review by rubenaprikyan — CHANGES_REQUESTED** [review]
lgtm!

File: apis/core-api/src/api/middlewares/check-fan-club-role/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const hasRole = await FanClubService.isMemberHasRole(
+      fanClubId,
+      ctx.user.userId,
+      roles
+    );
+
+    if (!hasRole) {
+      throw new AccessDeniedError({
+        errorCode: FanClubMemberErrorEnum.permissionDenied,
+        message: "You don't have a valid role to complete this action.",
```
[the remark this answers is not in the document]
Comment (rado196) [reply apis/core-api/src/api/middlewares/check-fan-club-role/index.ts:25]: will change

File: apis/core-api/src/core/services/FanClubService.ts [hunk 2]
```diff
@@ -135,6 +143,9 @@ class FanClubService extends BaseService {
       paranoid: false,
     });
 
+    const roleId = await this.getRoleId(role);
+    console.log({ roleId, role });
```
[the remark this answers is not in the document]
Comment (rado196) [reply apis/core-api/src/core/services/FanClubService.ts]: oh, will remove

**Review by rubenaprikyan — APPROVED** [review]

File: apis/core-api/src/api/middlewares/check-fan-club-role/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    const fanClubId = ctx.request.params.id;
+    const hasRole = await FanClubService.isMemberHasRole(
+      fanClubId,
+      ctx.user.userId,
+      roles
+    );
+
+    if (!hasRole) {
```
[the remark this answers is not in the document]
Comment (rado196) [reply apis/core-api/src/api/middlewares/check-fan-club-role/index.ts:22]: I think both namings are correct :)

**Pull request merged**